lib/hawkular/inventory/entities_v4.rb

```diff
@@ -82,6 +82,16 @@ def initialize(hash)
       @_hash = hash.dup
     end
 
+    def children_by_type(type, recursive = false)
+      fail Hawkular::ArgumentError 'Resource tree not loaded, load it by calling resource_tree' if @children.nil?
+      selected_children = []
+      @children.each do |child|
+        selected_children << child if child.type.id == type
+        selected_children.push(*child.children_by_type(type, recursive)) if recursive
+      end
+      selected_children
```

Comment:
No way to do it iteractively with a `flat_map`? Something like this might work:

```ruby
selected_children = @children.select { |x| x.type.id == type }

if recursive
  selected_children.flat_map! { |x| x.children_by_type(type, recursive) }
end

selected_children
```

Reply:
Will check, but i don't think that would work.

`selected_children = @children.select { |x| x.type.id == type }`
This will select the children with the required `type`, this is good.

`selected_children.flat_map! { |x| x.children_by_type(type, recursive) }`
This will only iterate over the selected children, this is bad, I want to check in all the children.

Reply:
```ruby
selected_children = @children.select { |x| x.type.id == type }
if recursive
  selected_children.push(*@children.flat_map { |x| x.children_by_type(type, recursive) })
end
```
Is this better readable?